source/extensions/common/proxy_protocol/proxy_protocol_header.cc

```diff
@@ -111,12 +111,26 @@ void generateV2Header(const Network::Address::Ip& source_address,
 }
 
 bool generateV2Header(const Network::ProxyProtocolData& proxy_proto_data, Buffer::Instance& out,
-                      bool pass_all_tlvs, const absl::flat_hash_set<uint8_t>& pass_through_tlvs) {
-  uint64_t extension_length = 0;
-  for (auto&& tlv : proxy_proto_data.tlv_vector_) {
+                      bool pass_all_tlvs, const absl::flat_hash_set<uint8_t>& pass_through_tlvs,
+                      const std::unordered_map<uint8_t, std::vector<unsigned char>>& custom_tlvs) {
+  std::vector<Envoy::Network::ProxyProtocolTLV> combined_tlv_vector_;
+  for (const auto& [type, value] : custom_tlvs) {
+    combined_tlv_vector_.push_back({type, {value.begin(), value.end()}});
```

Comment:
should we do the check if already exists here too?

Reply:
Checking whether the custom TLV key already exists in the combined_tlv_vector_ variable? That should be empty after initialization on line 116.

Reply:
i was thinking that a user might have the same type twice in the protobuf; but that's a control-plane bug. so i'm ok with this as is

Reply:
Ah I see what you mean. We handled that edge case in the previous implementation by skipping duplicate keys. Agreed it would be a control plane bug, but I think we can still be handle those edge case in the buildCustomTLVs code.

Reply:
Handled those edge cases in https://github.com/timflannagan/envoy/pull/1/commits/8560b3dec09b54945f654a9a4c9dcb020611630d and added a unit test in https://github.com/timflannagan/envoy/pull/1/commits/26217fd1d743dcceb3409e430affc0fd32afa387.